openedx/adg/common/djangoapps/user_authn_override/views/registration_form.py

```diff
@@ -0,0 +1,161 @@
+"""
+Overridden RegistrationFormFactory from openedx.core.djangoapps.user_authn.views.registration_form.py
+"""
+
+
+from django.utils.translation import ugettext as _
+
+from edxmako.shortcuts import marketing_link
+from openedx.adg.constants import ADG_BUSINESS_LINES, CITIES_OF_KSA
+from openedx.core.djangoapps.user_api import accounts
+from openedx.core.djangoapps.user_authn.views.registration_form import (
+    RegistrationFormFactory as CoreRegistrationFormFactory
+)
+from openedx.core.djangolib.markup import HTML, Text
+
+
+class RegistrationFormFactory(CoreRegistrationFormFactory):
+    """
+    Overridden openedx RegistrationFormFactory to avoid core changes.
+    """
+
+    EXTRA_FIELDS = CoreRegistrationFormFactory.EXTRA_FIELDS + ["is_adg_employee"]
+
+    def _add_username_field(self, form_desc, required=True):
+        """
+        Add a username field to a form description.
+
+        Arguments:
+            form_desc: A form description
+
+        Keyword Arguments:
+            required (bool): Whether this field is required; defaults to True
+        """
+
+        # Translators: This label appears above a field on the registration form
+        # meant to hold the user's public username.
+
+        username_instructions = _(
+            # Translators: These instructions appear on the registration form, immediately
+            # below a field meant to hold the user's public username.
+            u"The name that will identify you in your courses. "
+            u"It cannot be changed later."
+        )
+        form_desc.add_field(
+            "username",
+            label=_(u"Username"),
+            instructions=username_instructions,
+            restrictions={
+                "min_length": accounts.USERNAME_MIN_LENGTH,
+                "max_length": accounts.USERNAME_MAX_LENGTH,
+            },
+            required=required
+        )
+
+    def _add_is_adg_employee_field(self, form_desc, required=False):
```

Comment:
Shouldn't these be added automatically through EXTRA_FIELDS?

Reply:
How? without writing anything? We need to provide field handlers and then the EXTRA_FIELDS are used to call these handlers based on REGISTRATION_EXTRA_FIELDS setting.